src/Utils/DEM.jl

```diff
@@ -0,0 +1,77 @@
+
+
+abstract type DEM end
+
+############# DEM Interface begin ######################################
+
+"""
+    getCoordinate(dem::T,index::Tuple{Integer,Integer}) where T <: DEM
+
+    Get the coordinate for a certain index in the DEM. 
+    (All the DEM's are stored as a matrix of heights)
+"""
+function getCoordinate(dem::T,index::Tuple{Integer,Integer}) where T <: DEM
+     ErrorException("Not implemented for the following DEM type, $T")
+end
+
+
+"""
+    getIndex(dem::T,coordinate::Tuple{Real,Real}) where T <: DEM
+
+    Get the index of the DEM corresponding to the coordinate. 
+    (All the DEM's are stored as a matrix of heights)
+"""
+function getIndex(dem::T,coordinate::Tuple{Real,Real}) where T <: DEM
+    ErrorException("Not implemented for the following DEM type, $T")
+end
+
+############# DEM Interface end ######################################
+
+
+"""
+    TandemxDEM <: DEM
+
+    A DEM implementation for Tandem-X DEM data.
+"""
+struct TandemxDEM <: DEM
+    heights::Array{Union{Missing,Float32},2}
+    coordinateOffset::Tuple{Float64,Float64} ## lat lon
+    coordinateSpacing::Tuple{Float64,Float64} ## lat lon
+end
+
+
+
+"""
+    loadTandemxDEM(tiffPath::String) -> TandemxDEM
+
+    Load a Tandem-X DEM tiff. The Tandem-X DEM's can be downloaded from https://download.geoservice.dlr.de/TDM90/.
+    Note: Invalid values are replaced with missing 
+"""
+function loadTandemxDEM(tiffPath::String)
```

Comment:
Why is the constructor not named after the structure? i.e. both called TandemxDEM

Reply:
As far as I now a constructor normally does not load data from files or from a database. This is therefore a normal function that loads the data and call the standard constructor in the end